test/cql-pytest/cassandra_tests/functions/operation_fcts_test.py

```diff
@@ -0,0 +1,819 @@
+                   row(1, 2, 0, 1, c_float(2.5).value, 0.5, 1, Decimal("2.5")))
+
+        assertRows(execute(cql, table, "SELECT a % d, b % d, c % d, d % d, e % d, f % d, g % d, h % d FROM %s WHERE a = 1 AND b = 2 AND c = 23 % 5"),
+                   row(1, 2, 3, 0, 1.5, 2.5, 3, Decimal("0.5")))
+
+        assertRows(execute(cql, table, "SELECT a % e, b % e, c % e, d % e, e % e, f % e, g % e, h % e FROM %s WHERE a = 1 AND b = 2 AND c = 23 % 5"),
+                   row(c_float(1.0).value, c_float(2.0).value, c_float(3.0).value, 4.0, c_float(0.0).value, 1.0, Decimal("1.5"), Decimal("3.0")))
+
+        assertRows(execute(cql, table, "SELECT a % f, b % f, c % f, d % f, e % f, f % f, g % f, h % f FROM %s WHERE a = 1 AND b = 2 AND c = 23 % 5"),
+                   row(1.0, 2.0, 3.0, 4.0, 5.5, 0.0, Decimal("0.5"), Decimal("2.0")))
+
+        assertRows(execute(cql, table, "SELECT a % g, b % g, c % g, d % g, e % g, f % g, g % g, h % g FROM %s WHERE a = 1 AND b = 2 AND c = 23 % 5"),
+                   row(1,2,3,4, Decimal("5.5"), Decimal("6.5"), 0, Decimal("1.5")))
+
+        assertRows(execute(cql, table, "SELECT a % h, b % h, c % h, d % h, e % h, f % h, g % h, h % h FROM %s WHERE a = 1 AND b = 2 AND c = 23 % 5"),
+                   row(Decimal("1.0"),
+                       Decimal("2.0"),
+                       Decimal("3.0"),
+                       Decimal("4.0"),
+                       Decimal("5.5"),
+                       Decimal("6.5"),
+                       Decimal("7"),
+                       Decimal("0.0")))
+
+        # Test negation
+
+        with original_column_names(cql) as cql:
+            assertColumnNames(execute(cql, table, "SELECT -a, -b, -c, -d, -e, -f, -g, -h FROM %s WHERE a = 1 AND b = 2"),
+                          "-a", "-b", "-c", "-d", "-e", "-f", "-g", "-h")
+
+        assertRows(execute(cql, table, "SELECT -a, -b, -c, -d, -e, -f, -g, -h FROM %s WHERE a = 1 AND b = 2"),
+                   row(-1, -2, -3, -4, c_float(-5.5).value, -6.5, -7, Decimal("-8.5")))
+
+        # Test with null
+        execute(cql, table, "UPDATE %s SET d = ? WHERE a = ? AND b = ? AND c = ?", None, 1, 2, 3)
+        assertRows(execute(cql, table, "SELECT a + d, b + d, c + d, d + d, e + d, f + d, g + d, h + d FROM %s WHERE a = 1 AND b = 2"),
+                   row(None, None, None, None, None, None, None, None))
+
+@pytest.mark.xfail(reason="#2693")
+def testModuloWithDecimals(cql, test_keyspace):
+    with create_table(cql, test_keyspace, "(numerator decimal, dec_mod decimal, int_mod int, bigint_mod bigint, PRIMARY KEY(numerator, dec_mod))") as table:
+        execute(cql, table, "INSERT INTO %s (numerator, dec_mod, int_mod, bigint_mod) VALUES (123456789112345678921234567893123456, 2, 2, 2)")
+
+        assertRows(execute(cql, table, "SELECT numerator % dec_mod, numerator % int_mod, numerator % bigint_mod from %s"),
+                   row(Decimal("0"), Decimal("0.0"), Decimal("0.0")))
```

Comment:
Can we slightly modify the test to assert for a result other than 0? Because what if there was an error and modulo always returned 0?

Reply:
This is the exact logic that Cassandra's test used in their test, I didn't modify it.
Often I don't like Cassandra's tests and add new ones for things I think they missed  - but they they are in test/cql-pytest outside the cassandra_tests subdirectory. I don't want to do this later (as you noted, this feature isn't something we think is very important or plan to implement soon, so there's a limit to how much effort I'm willing to put into it).